addons/math-in-inputs/userscript.js

```diff
@@ -0,0 +1,72 @@
+export default async function ({ addon }) {
+  function getRegexFromSettings() {
+    let textInInputs = addon.settings.get("textInInputs");
+    return textInInputs ? /^[0-9a-z+\-*^/().]+$/ : /^[0-9+\-*^/().]+$/;
+  }
+
+  // I 100% stole this part of the code from "editor-number-arrow-keys"
+  const isSupportedElement = (el) => {
+    if (el.classList.contains("blocklyHtmlInput")) return true;
+    else if (el.matches(".mediaRecorderPopupContent input[type=number]")) {
+      // Number inputs in `mediarecorder` addon modal
+      return true;
+    } else if (el.className.includes("input_input-form_")) {
+      if (el.matches("[class*=sprite-info_sprite-info_] [class*=input_input-small_]")) {
+      }
+      // Sprite X/Y coordinates, size and direction (excludes sprite name)
+      return true;
+    } else if (el.matches("[class*=paint-editor_editor-container-top_] input[type=number]")) {
+      // Number inputs in costume editor (note that browsers already provide up/down clickable buttons for these)
+      return true;
+    } else return false;
+  };
+
+  function power(match, x, y) {
+    return `Math.pow(${x}, ${y})`;
+  }
+
+  function parseMath(value) {
+    if (!/^[0-9+\-*/().^]+$/.test(value)) {
+      return value;
+    }
+
+    const processedValue = value.replace(/(\d+)\^(\d+)/g, power);
+
+    try {
+      return eval(processedValue) || 0;
```

Comment:
Whoever is reading this, we will not use eval. But it's fine while this stays as a draft.

Reply:
Ah darn I thought the regex check would be good enough, ill make a custom parser

Reply:
I did in fact not make it but its done